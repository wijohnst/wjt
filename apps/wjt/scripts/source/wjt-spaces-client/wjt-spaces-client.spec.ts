import { WjtSpacesClient, WJT_SPACES_BUCKET_ID } from './wjt-spaces-client';

import { S3Client, ListObjectsV2CommandOutput } from '@aws-sdk/client-s3';

jest.mock('@aws-sdk/client-s3');

describe('WjtSpacesClient', () => {
  let sut: WjtSpacesClient;

  beforeEach(() => {
    const s3Client = new S3Client({});
    sut = new WjtSpacesClient(s3Client);
  });

  test('should be defined', () => {
    expect(WjtSpacesClient).toBeDefined();
  });

  describe('getBucketId', () => {
    test('should return the bucketId', () => {
      expect(sut.getBucketId()).toEqual(WJT_SPACES_BUCKET_ID);
    });
  });

  describe('getBucketContents', () => {
    test('should return the bucket contents', async () => {
      const sut: WjtSpacesClient = new WjtSpacesClient({
        send: jest.fn().mockResolvedValue({
          Contents: [
            {
              Key: 'image-1.jpg',
              LastModified: new Date('2024-12-25'),
              ETag: 'etag',
              Size: 123,
              StorageClass: 'STANDARD',
            },
          ],
        } as ListObjectsV2CommandOutput),
      } as unknown as S3Client);

      const response = await sut.getBucketContents();

      expect(response).toBeDefined();
      expect(response).toMatchSnapshot();
    });
  });
});
