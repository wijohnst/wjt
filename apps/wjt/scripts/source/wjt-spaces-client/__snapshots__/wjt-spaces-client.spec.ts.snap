// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WjtSpacesClient getBucketContents should return the bucket contents 1`] = `
[
  {
    "ETag": "etag",
    "Key": "image-1.jpg",
    "LastModified": 2024-12-25T00:00:00.000Z,
    "Size": 123,
    "StorageClass": "STANDARD",
  },
]
`;
